### PR TITLE
28 implement refresh token

### PR DESCRIPTION
**Pull request by tsiguenz**
* To test this PR you can change the expired time in auth.service / createTokens.
* The access token can be a really low value (I test with 10s) to force the utilisation of the refresh token.
* if you find a better solution to the reload in the main.ts don't hesitate to let a comment

### CODE REVIEW

**Review by gmorange42 — CHANGES_REQUESTED** [review]
Nice !

File: backend/src/auth/auth.service.ts [hunk 1]
```diff
@@ -33,7 +35,9 @@ export class AuthService {
         hash: hash
       }
     });
-    return this.createJwt(user.id);
+    const tokens = await this.createTokens(user.id);
+    await this.updateRefreshToken(user.id, tokens.refresh_token);
+    return tokens;
```
Comment (Crysicia) [inline backend/src/auth/auth.service.ts:40]: Might be extracted in a function, you repeat the same 3 lines multiple times

File: backend/src/auth/auth.service.ts [hunk 2]
```diff
@@ -54,22 +58,64 @@ export class AuthService {
         message: 'Two factor code required',
         id: await this.twoFa.generateTwoFaId(user.id)
       };
-    return this.createJwt(user.id);
+    const tokens = await this.createTokens(user.id);
+    await this.updateRefreshToken(user.id, tokens.refresh_token);
+    return tokens;
```
Comment (Crysicia) [inline backend/src/auth/auth.service.ts:63]: Here

File: backend/src/auth/auth.service.ts [hunk 3]
```diff
@@ -54,22 +58,64 @@ export class AuthService {
...
+    if (!user || !user.refreshToken)
+      throw new ForbiddenException('Access Denied');
+    const refreshTokenMatches = await argon.verify(
+      user.refreshToken,
+      refreshToken
+    );
+    if (!refreshTokenMatches) throw new ForbiddenException('Access Denied');
+    const tokens = await this.createTokens(user.id);
+    await this.updateRefreshToken(user.id, tokens.refresh_token);
+    return tokens;
```
Comment (Crysicia) [inline backend/src/auth/auth.service.ts:100]: And here

**Comment by Crysicia** [comment]
Nice work.
Does it work with WebSockets ?

**Review by gmorange42 — APPROVED** [review]
All looks good, GG.

**Pull request merged**